rpcserver.go

```diff
@@ -4512,16 +4512,65 @@ func (r *rpcServer) FeeReport(ctx context.Context,
 // 0.000001, or 0.0001%.
 const minFeeRate = 1e-6
 
+// checkMaxHtlc errors if the user tries to update their channel policy with
+// a max HTLC value that is too high or too low for a given list of channels.
+func checkMaxHtlc(maxHtlc uint64, channels []*channeldb.OpenChannel) error {
+	var (
+		minSupportedHtlc lnwire.MilliSatoshi
+		maxSupportedHtlc = lnwire.MilliSatoshi(math.MaxUint64)
+	)
+
+	for _, ch := range channels {
```

Comment:
I think we should be applying the constraint to each channel individually. If there are two channels with `min_htlc`s of 4 and 6 and we're proposing a `max_htlc` of 5, we should not allow this `max_htlc` for the the channel with the higher `min_htlc`, while the current diff allows it. Doing so will also allow us to include the channel point in the error as well.

Reply:
Discussed, hopefully the code is more clear now.